### ISSUE TITLE
iOS deferred deep linking is not working

### ISSUE THREAD

**Issue by RobertoNegro** [BranchMetrics/react-native-branch-deep-linking-attribution#627]
We're currently implementing Branch.io in our app, but we've encountered some problems in the deferred deep linking on iOS.

In general, Branch.io links are working well on Android, both for deferred deep links but also for deep links clicked with the app already installed.
In iOS, instead, they work well only if the app is already installed. Otherwise, the link opens the App Store, but after the app is installed the information from the link are not available (in the subscribe i just obtain {"+clicked_branch_link": false, "+is_first_session": false, "+rn_cached_initial_event": true}).

I've tried using getFirstReferringParams, but even if the documentation says that it should be an asynchronous function, if I use await React gives me an error. If I remove the await, in fact, the object returned from that function is not a promise, it's just a plain empty object but with no useful attributes. The same behaviour is obtained also with getLatestReferringParams, even if from the subscribe callback I can obtain the parameters from the deep link that I've clicked on, when the app is already installed. 

I've checked the keys, and they should be correctly set: I'm using the live key and it's correctly set both in the Info.plist (branch_key) and also in the branch.json. In branch.json debugMode and useTestInstance are set to false.

Universal Links should be fine as well: If I long press a link, it says "Open with X".

I've also tried different "URI Scheme Deep Link Mode" from branch.io configuration page to check if maybe it was a problem of deep linking, without any success.

I can't understand what could be wrong from my side, it looks like the deferred linking is not working at all. 

Thank you

**Comment by oscarjes** [comment]
Exactly the same issue. Did you find a solution @RobertoNegro ?

**Comment by RobertoNegro** [comment]
No, sadly after many attempts I had to change library to be able to successfully manage deferred deep links.

**Comment by oscarjes** [comment]
> > No, sadly after many attempts I had to change library to be able to successfully manage deferred deep links.

Thanks @RobertoNegro. Would you mind sharing what other library you ended up using?

**Comment by RobertoNegro** [comment]
> > No, sadly after many attempts I had to change library to be able to successfully manage deferred deep links.
> 
> Thanks @RobertoNegro. Would you mind sharing what other library you ended up using?

I'm currently using AppsFlyer. They have a ReactNative library and the replacement was almost painless. In my case, I needed only deferred deep linking and direct deep linking, I was not looking for statistics, campaigns, etc, so I'm not sure if the features of Branch are all covered in AppsFlyer. But, in any case, they have plenty of features as well (maybe more, maybe less, I really don't know), and at least the development is active and their support is really good (I had a misunderstanding with the usage of their library and they helped me to solve it).

**Comment by Meet-JS** [comment]
Can we have an update on this please? I'm using v5.4.0.
